src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthTests.cs

```diff
@@ -0,0 +1,453 @@
+
+        [Fact]
+        public async Task NumOfBytesAsLiteralConst()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+        
+        Buffer.BlockCopy(src, 0, dst, 0, 8);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(src, 0, dst, 0, 8)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NumOfBytesAsMultipleDeclarations()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+        int test = 4, numOfBytes = src.Length;
+
+        Buffer.BlockCopy(src, 0, dst, 0, [|numOfBytes|]);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+        Dim test = 4, numOfBytes = src.Length
+
+        Buffer.BlockCopy(src, 0, dst, 0, [|numOfBytes|])
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NumOfBytesAsConstLiteral()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+        int numOfBytes = 8;
+
+        Buffer.BlockCopy(src, 0, dst, 0, numOfBytes);
+    }
+}
+");
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+        Dim numOfBytes = 8
+
+        Buffer.BlockCopy(src, 0, dst, 0, numOfBytes)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NumOfBytesAsClassConstProperty()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    private const int field = 3;
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+
+        Buffer.BlockCopy(src, 0, dst, 0, Program.field);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Private Const field As Integer = 8
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(src, 0, dst, 0, Program.field)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NumOfBytesAsMethodInvocation()
```

Comment:
Add another test case here to test `Buffer.BlockCopy(new int[] {1, 2, 3, 4}, 0, new int[] {5, 6, 7, 8}, 0, someLength)` ?